### PR TITLE
Added copy-file api and writeOcTsx

### PR DESCRIPTION
**Pull request by rishabh-ti**

### CODE REVIEW

File: src/components/sidebar/Sidebar.tsx [hunk 1]
```diff
@@ -23,6 +23,7 @@ import {
 } from '~core/selectors/customComponents'
 import useDispatch from '~hooks/useDispatch'
 import API from '~custom-components/api'
+import components from '~core/models/components'
```
Comment (vinayak-sachdeva) [inline src/components/sidebar/Sidebar.tsx]: redundant

File: src/components/sidebar/Sidebar.tsx [hunk 2]
```diff
@@ -59,6 +60,7 @@ const Menu = () => {
           path: newComponentsList[component],
         })
       })
+      const response = await API.post('/copy-file', customComponents)
```
Comment (vinayak-sachdeva) [inline src/components/sidebar/Sidebar.tsx]: shouldn't we use newComponentList?

**Review by vinayak-sachdeva — CHANGES_REQUESTED** [review]

File: src/components/sidebar/Sidebar.tsx (hunk 1 above)
Comment (rishabh-ti) [reply src/components/sidebar/Sidebar.tsx]: Removed

File: src/components/sidebar/Sidebar.tsx (hunk 2 above)
Comment (rishabh-ti) [reply src/components/sidebar/Sidebar.tsx]: Using newComponentList

**Pull request merged**